module_analysis/3_calc_prob.py

```diff
@@ -0,0 +1,170 @@
+# $$
+#
+# A similar probability can be calculated for core genes co-expressed with other core genes.
+
+# %load_ext autoreload
+# %autoreload 2
+# %matplotlib inline
+import os
+import pandas as pd
+import numpy as np
+from core_acc_modules import utils, paths
+
+# User params
+method = "affinity"
+
+# +
+# Import module memberships -- import annotated df
+pao1_membership_filename = os.path.join(
+    paths.LOCAL_DATA_DIR, f"pao1_gene_module_annotated_{method}.tsv"
+)
+pa14_membership_filename = os.path.join(
+    paths.LOCAL_DATA_DIR, f"pa14_gene_module_annotated_{method}.tsv"
+)
+
+pao1_membership = pd.read_csv(pao1_membership_filename, sep="\t", index_col=0, header=0)
+pa14_membership = pd.read_csv(pa14_membership_filename, sep="\t", index_col=0, header=0)
+# -
+
+pao1_membership.head()
+
+pa14_membership.head()
+
+# ## Get core/accessory annotations
+
+# +
+# Read in expression data
+pao1_expression_filename = paths.PAO1_COMPENDIUM
+pa14_expression_filename = paths.PA14_COMPENDIUM
+
+pao1_expression = pd.read_csv(pao1_expression_filename, sep="\t", index_col=0, header=0)
+pa14_expression = pd.read_csv(pa14_expression_filename, sep="\t", index_col=0, header=0)
+
+# +
+pao1_annot_filename = paths.GENE_PAO1_ANNOT
+pa14_annot_filename = paths.GENE_PA14_ANNOT
+
+core_acc_dict = utils.get_my_core_acc_genes(
+    pao1_annot_filename, pa14_annot_filename, pao1_expression, pa14_expression
+)
+# -
+
+pao1_core = core_acc_dict["core_pao1"]
+pa14_core = core_acc_dict["core_pa14"]
+pao1_acc = core_acc_dict["acc_pao1"]
+pa14_acc = core_acc_dict["acc_pa14"]
+
+# ## Calculate likelihood
+# $Pr(\text{gene x in acc module}\cap \text{gene x is acc gene})$ is the number of accessory genes in accessory modules
+
+# Get subset of df
+pao1_subset = pao1_membership[["module id", "module label", "num acc in module"]]
+pao1_subset = pao1_subset.set_index("module id")
+pao1_subset = pao1_subset.drop_duplicates()
+
+pa14_subset = pa14_membership[["module id", "module label", "num acc in module"]]
+pa14_subset = pa14_subset.set_index("module id")
+pa14_subset = pa14_subset.drop_duplicates()
+
+# Get number of accessory genes in "mostly accessory" modules
+num_acc_gene_in_acc_mod_pao1 = pao1_subset.loc[
+    pao1_subset["module label"] == "mostly accessory", "num acc in module"
+].sum()
+num_acc_gene_in_acc_mod_pa14 = pa14_subset.loc[
+    pa14_subset["module label"] == "mostly accessory", "num acc in module"
+].sum()
+
+# Currently there are no mostly core modules, so this is a placeholder
+# Get number of core genes in "mostly core" modules
+"""num_core_gene_in_core_mod_pao1 = pao1_subset.loc[
+    pao1_subset["module label"] == "mostly core", "num core in module"
+].sum()
+num_core_gene_in_core_mod_pa14 = pa14_subset.loc[
+    pa14_subset["module label"] == "mostly core", "num core in module"
+].sum()"""
+
+lik_pao1_acc = num_acc_gene_in_acc_mod_pao1 / (len(pao1_core) + len(pao1_acc))
```

Comment:
I think you could just do this in one step if you want, since the total numbers of genes will cancel out. Here's the logic for how I got there (see if you agree with it), based on the formula at the top of the notebook:

![Image from iOS (5)](https://user-images.githubusercontent.com/2345877/125828635-bcb03b26-bb63-424b-a8fc-dbc09c676c62.jpg)

This should give the same answer as what you have, but might save you a few lines of code - unless you want to keep looking at the prior distribution as part of the notebook (which I'm totally fine with).

Reply:
NICE MATH! I'm ok keeping it this way so its more intuitive for future me:)